### ISSUE TITLE
Android GoBack with forcefully close the app even if the keyboard is up

### ISSUE THREAD

**Issue by alexballas** [fyne-io/fyne#4257]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Hello,
Since v2.4.0 GoBack action will forcefully close the app even if they keyboard is up.

### How to reproduce

1) Create an app with a textbox
2) Select the textbox. The keyboard appears.
3) Issue a back action

Expected:
The keyboard should collapse.

What happens:
The app exits.

### Example code

```go
package main

import (
	"log"

	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	myApp := app.New()
	myWindow := myApp.NewWindow("Entry Widget")

	input := widget.NewEntry()
	input.SetPlaceHolder("Enter text...")

	content := container.NewVBox(input, widget.NewButton("Save", func() {
		log.Println("Content was:", input.Text)
	}))

	myWindow.SetContent(content)
	myWindow.ShowAndRun()
}
```

### Fyne version

v2.4.0

### Go compiler version

1.21.1

### Operating system and version

Ubuntu 18.04 (compiling)  / Android 10

**Comment by Jacalz** [comment]
It looks like you forgot to add the example code. Would you mind updating the description?

**Comment by alexballas** [comment]
Hello @Jacalz ,I copy pasted the example from https://developer.fyne.io/widget/entry
Any app with an entry widget should do (and assuming you don't do any special handling with `w.Canvas().SetOnTypedKey`)

**Comment by andydotxyz** [comment]
Fixed on develop and will be in v2.4.3
